packages/analytics/analytics-chart/src/components/chart-plugins/HighlightPlugin.ts

```diff
@@ -54,7 +60,8 @@ export const highlightPlugin: HighlightPlugin = {
       chart.options.hover = { mode: null }
     } else {
       delete this.clickedElements
-      chart.options.hover = this.previousHoverOption
+      chart.options.hover = this.previousHoverOption || chart.options.hover
```

Comment:
this.previousHoverOption will only ever be the cached hover option that was saved in the previous if block, in order to prevent hover interactions while the tooltip is "locked" 

Checking it for undefined is fine, I'm not sure what else to check it for, it's only this plugin that is aware of "previousHoverOption"

Reply:
previousHoverOption only exists on the HighlightPlugin, will not be set or modified by anything else

<img width="669" alt="image" src="https://github.com/Kong/public-ui-components/assets/6026470/9da9733a-a905-475c-b25e-2da6369780d9">

Reply:
I know: I'm saying that when you set `chart.options.hover`, I'm _pretty_ sure that triggers ChartJS to do work to rebuild some internal internal structures.  It would be better to avoid saying `chart.options.hover = <anything>` unless you have to.

I won't block the PR over it; this is mostly a suggestion for optimization in general.

Reply:
Ohh I see what you mean. 

I don't think the options on this chart instance are actually reactive in the way you think. I don't think there's any other way to disable the chart's hovering behaviour though